scripts/recall-last-item.lua

```diff
@@ -1,18 +1,34 @@
 local util = require("scripts.util")
 
+local MAX_HISTORY_SIZE = 10
+
 --- @param e EventData.CustomInputEvent
 local function on_recall_last_item(e)
   local player = game.get_player(e.player_index)
   if not player then
     return
   end
 
-  local last_item = storage.last_item[e.player_index]
-  if not last_item then
+  local history = storage.item_history[e.player_index]
+  local index = storage.history_index[e.player_index]
+
+  if not (history and index) then
     return
   end
 
-  util.set_cursor(player, last_item)
+  local cursor_item = util.get_cursor_item(player)
```

Comment:
This whole block is causing what I consider to be unintuitive behavior:

https://github.com/user-attachments/assets/63b49056-4f8f-42d9-ab08-e72c0fe2493b

I would expect it to cycle through all four items.

Reply:
fixed, but it made code a bit awkward (at least in my implementation)